drivers/eeprom/eeprom_native_posix.c

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright (c) 2019 Vestas Wind Systems A/S
+ *
+ * Heavily based on flash_native_posix.c, which is:
+ * Copyright (c) 2019 Jan Van Winkel <jan.van_winkel@dxplore.eu>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <drivers/eeprom.h>
+
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <sys/mman.h>
+#include <fcntl.h>
+#include <string.h>
+#include <errno.h>
+
+#include "cmdline.h"
+#include "soc.h"
+
+#define LOG_LEVEL CONFIG_EEPROM_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(eeprom_native_posix);
+
+static const char default_eeprom_path[] = "eeprom.bin";
+
+struct eeprom_native_posix_data {
+	struct k_mutex lock;
+	const char *path;
+	int fd;
+	u8_t *eeprom;
+	bool init_called;
+};
+
+struct eeprom_native_posix_config {
+	size_t size;
+	bool readonly;
+};
+
+#define DEV_NAME(dev) ((dev)->config->name)
+#define DEV_CONFIG(dev) ((dev)->config->config_info)
+#define DEV_DATA(dev) \
+	((struct eeprom_native_posix_data *const)(dev)->driver_data)
+
+static int eeprom_native_posix_read(struct device *dev, off_t offset,
+				    void *buf, size_t len)
+{
+	struct eeprom_native_posix_data *const data = DEV_DATA(dev);
+	const struct eeprom_native_posix_config *config = DEV_CONFIG(dev);
+
+	if (!len) {
+		return 0;
+	}
+
+	if ((offset + len) > config->size) {
+		LOG_WRN("attempt to read past device boundary");
+		return -EINVAL;
+	}
+
+	if (data->eeprom == MAP_FAILED) {
+		LOG_ERR("no EEPROM device mapped");
+		return -EIO;
+	}
+
+	k_mutex_lock(&data->lock, K_FOREVER);
```

Comment:
Just a silly comment: In native_posix this would be superstitious :),
the memcpy finishes in 0 simulated time and cannot be interrupted.
It does not harm to use a mutex, but it does not help either.

Reply:
@aescolar Thanks. I have removed the mutex.